### PR TITLE
XCAssets: add support for `symbolset`

### LINKED ISSUE #640: SF Symbols support
It would be great if SwiftGen could generate static enum cases for the new SF Symbols that arrive in iOS 13.

### LINKED ISSUE #767: XCAssets: Custom SFSymbols support (aka symbolset)
Would there be interest in supporting custom SFSymbols as described here?
https://developer.apple.com/documentation/uikit/uiimage/creating_custom_symbol_images_for_your_app

Works quite similar to imagesets.

### PR DESCRIPTION
**Pull request by djbe**
Fixes #767.

### CODE REVIEW

**Comment by SwiftGen-Eve** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

Hey 👋 I'm Eve, the friendly bot watching over SwiftGen 🤖

Thanks a lot for your contribution!



---

Seems like everything is in order 👍 You did a good job here! 🤝

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: SwiftGen.playground/Pages/XCAssets-Demo.xcplaygroundpage/Sources/Implementation Details.swift [hunk 1]
```diff
@@ -160,3 +160,38 @@ public extension ImageAsset.Image {
...
+  public typealias Image = UIImage
+
+  public var image: Image {
+    #if os(iOS) || os(tvOS)
+    let image = Image(named: name, in: bundle, compatibleWith: nil)
+    #elseif os(watchOS)
+    let image = Image(named: name)
+    #endif
+    guard let result = image else {
+      fatalError("Unable to load image asset named \(name).")
```
Comment (Lutzifer) [inline SwiftGen.playground/Pages/XCAssets-Demo.xcplaygroundpage/Sources/Implementation Details.swift]:
```suggestion
      fatalError("Unable to load symbol asset named \(name).")
```

File: SwiftGen.playground/Pages/XCAssets-Demo.xcplaygroundpage/Sources/Implementation Details.swift [hunk 2]
```diff
@@ -160,3 +160,38 @@ public extension ImageAsset.Image {
...
+    }
+    return result
+  }
+
+  @available(iOS 13.0, tvOS 13.0, watchOS 6.0, *)
+  public func image(with configuration: Configuration) -> Image {
+    guard let result = Image(named: name, in: bundle, with: configuration) else {
+      fatalError("Unable to load image asset named \(name).")
```
Comment (Lutzifer) [inline SwiftGen.playground/Pages/XCAssets-Demo.xcplaygroundpage/Sources/Implementation Details.swift]:
```suggestion
      fatalError("Unable to load symbol asset named \(name).")
```

File: templates/xcassets/swift4.stencil [hunk 3]
```diff
@@ -247,6 +255,40 @@
...
+  
+  {{accessModifier}} var image: Image {
+    let bundle = {{param.bundle|default:"BundleToken.bundle"}}
+    #if os(iOS) || os(tvOS)
+    let image = Image(named: name, in: bundle, compatibleWith: nil)
+    #elseif os(watchOS)
+    let image = Image(named: name)
+    #endif
+    guard let result = image else {
+      fatalError("Unable to load image asset named \(name).")
```
Comment (Lutzifer) [inline templates/xcassets/swift4.stencil]:
```suggestion
      fatalError("Unable to load symbol asset named \(name).")
```

File: templates/xcassets/swift4.stencil [hunk 4]
```diff
@@ -247,6 +255,40 @@
...
+    }
+    return result
+  }
+
+  @available(iOS 13.0, tvOS 13.0, watchOS 6.0, *)
+  {{accessModifier}} func image(with configuration: Configuration) -> Image {
+    let bundle = {{param.bundle|default:"BundleToken.bundle"}}
+    guard let result = Image(named: name, in: bundle, with: configuration) else {
+      fatalError("Unable to load image asset named \(name).")
```
Comment (Lutzifer) [inline templates/xcassets/swift4.stencil:285]:
```suggestion
      fatalError("Unable to load symbol asset named \(name).")
```

**Comment by Lutzifer** [comment]
@djbe

Thanks a lot for implementing this feature.

I tested it using " mint run SwiftGen/SwiftGen@feature/assets-symbolset swiftgen" on our codebase and everything works beautifully.

Other than the "image" in the fatal error messages (I did not mark all of them) I did not see anything unusual/problematic in the code.

**Review by Lutzifer — APPROVED** [review]
The fatal error messages can be improved by replacing "image" with "symbol"

**Pull request merged**